flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/AbstractBroadcastStateTransformation.java

```diff
@@ -103,7 +103,7 @@ public void setChainingStrategy(ChainingStrategy chainingStrategy) {
     }
 
     @Override
-    public List<Transformation<?>> getTransitivePredecessors() {
+    protected List<Transformation<?>> getTransitivePredecessorsInternal() {
```

Comment:
This method, and a few other methods like in TwoInputTransformation and UnionTransformation, need to be fixed too. 
And it's better to introduce a test for each of them. Maybe we can introduce a separate test class `GetTransitivePredecessorsTest` to host them all.